### PR TITLE
Support basic auth for OTLP output in TracePipeline

### PR DESCRIPTION
**Pull request by chrkl**
**Description**

This PR adds initial basic auth support for the OTLP output in TracePipelines. Handling secret references will be added in a separate PR.

Changes proposed in this pull request:

- Add basic auth support for TracePipeline OTLP output

**Related issue(s)**

#15829

### CODE REVIEW

File: components/telemetry-operator/controller/tracepipeline/resources_test.go [hunk 1]
```diff
@@ -13,12 +15,30 @@ var (
...
 	tracePipeline = v1alpha1.TracePipelineOutput{
-		Otlp: v1alpha1.OtlpOutput{
+		Otlp: &v1alpha1.OtlpOutput{
 			Endpoint: v1alpha1.ValueType{
 				Value: "localhost",
 			},
 		},
 	}
+
+	tracePipelineWithBaiscAuth = v1alpha1.TracePipelineOutput{
```
Comment (lindnerby) [inline components/telemetry-operator/controller/tracepipeline/resources_test.go:25]:
```suggestion
	tracePipelineWithBasicAuth = v1alpha1.TracePipelineOutput{
```

File: components/telemetry-operator/controller/tracepipeline/resources_test.go [hunk 2]
```diff
@@ -30,6 +50,40 @@ func TestMakeConfigMap(t *testing.T) {
 	require.NotEmpty(t, cm.Data[configMapKey])
 }
 
+func TestMakeConfigMapWithBasicAuth(t *testing.T) {
+	cm := makeConfigMap(config, tracePipelineWithBaiscAuth)
```
Comment (lindnerby) [inline components/telemetry-operator/controller/tracepipeline/resources_test.go:54]:
```suggestion
	cm := makeConfigMap(config, tracePipelineWithBasicAuth)
```

File: components/telemetry-operator/controller/tracepipeline/resources_test.go [hunk 3]
```diff
@@ -30,6 +50,40 @@ func TestMakeConfigMap(t *testing.T) {
...
+	require.NotNil(t, secret)
+	require.Equal(t, secret.Name, config.ResourceName)
+	require.Equal(t, secret.Namespace, config.CollectorNamespace)
+
+	// A TracePipeline without secrets should produce an empty secret
+	require.Empty(t, secret.Data)
+}
+
+func TestMakeSecretWithBasicAuth(t *testing.T) {
+	secret := makeSecret(config, tracePipelineWithBaiscAuth.Otlp)
```
Comment (lindnerby) [inline components/telemetry-operator/controller/tracepipeline/resources_test.go:76]:
```suggestion
	secret := makeSecret(config, tracePipelineWithBasicAuth.Otlp)
```

File: components/telemetry-operator/controller/tracepipeline/controller_test.go [hunk 4]
```diff
@@ -36,7 +36,7 @@ var _ = Describe("Deploying a TracePipeline", func() {
 			},
 			Spec: telemetryv1alpha1.TracePipelineSpec{
 				Output: telemetryv1alpha1.TracePipelineOutput{
-					Otlp: telemetryv1alpha1.OtlpOutput{
+					Otlp: &telemetryv1alpha1.OtlpOutput{
```
Comment (lindnerby) [inline components/telemetry-operator/controller/tracepipeline/controller_test.go:39]: Would probably make more sense to start testing here more than just the minimal setting and asserting that CM and Secret are nil and instead create as much as possible. What do you think?

File: components/telemetry-operator/controller/tracepipeline/resources.go [hunk 5]
```diff
@@ -13,11 +15,8 @@ import (
 )
 
 var (
-	podAnnotations = map[string]string{
-		"sidecar.istio.io/inject": "false",
-	}
-	replicas           = int32(1)
-	collectorResources = corev1.ResourceRequirements{
+	basicAuthHeaderVariable = "BASIC_AUTH_HEADER"
```
Comment (lindnerby) [inline components/telemetry-operator/controller/tracepipeline/resources.go:18]: Should be `const`

File: components/telemetry-operator/controller/tracepipeline/controller_test.go (hunk 4 above)
Comment (chrkl) [reply components/telemetry-operator/controller/tracepipeline/controller_test.go:39]: The concurrent test execution of the Ginkgo framework and our limitation to one TracePipeline make it hard to do any comprehensive tests here. So I would prefer to keep testing the content of the generated ConfigMaps, etc. in resources_test.go and limit this test on validating the creation of the Kubernetes objects. But indeed we could do more. I suggest to check for the correct metadata and owner references here.

**Review by lindnerby — APPROVED** [review]
